queries/c/locals.scm

```diff
@@ -1,25 +1,44 @@
 ;; Functions definitions
 (function_declarator
-  declarator: (identifier) @definition.function) @scope
+  declarator: (identifier) @definition.function)
 (preproc_function_def
   name: (identifier) @definition.macro) @scope
 
 (preproc_def
   name: (identifier) @definition.macro)
 (pointer_declarator
-  declarator: (identifier) @definition.variable)
+  declarator: (identifier) @definition.var)
 (parameter_declaration
-  declarator: (identifier) @definition.variable)
+  declarator: (identifier) @definition.var)
 (init_declarator
-  declarator: (identifier) @definition.variable)
+  declarator: (identifier) @definition.var)
 (array_declarator
-  declarator: (identifier) @definition.variable)
+  declarator: (identifier) @definition.var)
 (declaration
-  declarator: (identifier) @definition.variable)
+  declarator: (identifier) @definition.var)
 (enum_specifier
   name: (*) @definition.type
   (enumerator_list
-    (enumerator name: (identifier) @definition.variable)))
+    (enumerator name: (identifier) @definition.var)))
+
+;; Parameters 
+; normal ones
+(parameter_list
+  (parameter_declaration 
+    declarator: (identifier) @definition.var))
+    
+; pointers
+(parameter_list
+  (parameter_declaration 
+    declarator: (pointer_declarator
+                  declarator: (identifier) @definition.var)))
+
+; pointers to pointers
+(parameter_list
+  (parameter_declaration 
+    declarator: (pointer_declarator
+                  declarator: (pointer_declarator
+                                declarator: (identifier) @definition.var))))
```

Comment:
The problem with C is those pointers, I think that we may want to use something like
```suggestion
(parameter_list
  (parameter_declaration 
    declarator: (.*) @declaration.var))
```
 And then let the user use it

Reply:
But then you will have the whole variable declaration, not only the type. Those to declarations strip away everything for the user (also possible const, volatile madness).

I admit that there's has to be a more elegant way to write this but if not repeating the nested structure won't hurt us to much (assuming that triple pointers are very rare).

Reply:
Maybe this shows that we may want to add a predicate which would take the first child matching a given query, something like `(submatch? @decl "((identifier) @declaration.var)"`

Reply:
Another, possibility would to use the `#strip!` instruction that they use in the treesitter repo:
```((parameter_list
  (parameter_declaration 
    declarator: (*) @definition.var))
    (#strip!  @definition.var "^.*\\s"))
```
Then one could only select the right most word of the query.

Reply:
Ok. I saw that the thing with the parameter decelerator is not necessary any more when the function block now belongs to the function scope. The query: 
```scheme
(pointer_declarator
  declarator: (identifier) @definition.var)
```
already defines the pointer correctly.